RobotFramework_UDS/UDSKeywords.py

```diff
@@ -1558,9 +1570,9 @@ def write_data_by_name(self, service_name = None, value = None, device_name = "d
 
         # Get service from name and verify the service is available
         diag_service_list = uds_device.diag_service_db.get_diag_service_by_name([service_name])
-        data_id = diag_service_list[0].request.parameters[1].coded_value
-
-        response = self.write_data_by_identifier(data_id, value, device_name)
+        dict_sub_service = { service_name: [sub_service] }
+        data_id = uds_device.diag_service_db.get_id_base_on_parameters_type(diag_service_list[0], dict_sub_service)
```

Comment:
double-check that `sub_service` is required when `service_name` contains data identifier as `TABLE-KEY` type